src/internal/packager2/layout/create.go

```diff
@@ -0,0 +1,467 @@
+	}
+
+	// Record the hostname of the package creation terminal.
+	// The error here is ignored because the hostname is not critical to the package creation.
+	hostname, _ := os.Hostname()
+	pkg.Build.Terminal = hostname
+
+	if pkg.IsInitConfig() {
+		pkg.Metadata.Version = config.CLIVersion
+	}
+
+	pkg.Build.Architecture = pkg.Metadata.Architecture
+
+	// Record the Zarf Version the CLI was built with.
+	pkg.Build.Version = config.CLIVersion
+
+	// Record the time of package creation.
+	pkg.Build.Timestamp = now.Format(time.RFC1123Z)
+
+	// Record the migrations that will be ran on the package.
+	pkg.Build.Migrations = []string{
+		deprecated.ScriptsToActionsMigrated,
+		deprecated.PluralizeSetVariable,
+	}
+
+	// Record the flavor of Zarf used to build this package (if any).
+	pkg.Build.Flavor = flavor
+
+	pkg.Build.RegistryOverrides = registryOverrides
+
+	// Record the latest version of Zarf without breaking changes to the package structure.
+	pkg.Build.LastNonBreakingVersion = deprecated.LastNonBreakingVersion
+
+	return pkg
+}
+
+func getChecksum(dirPath string) (string, string, error) {
+	checksumData := []string{}
+	err := filepath.Walk(dirPath, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		rel, err := filepath.Rel(dirPath, path)
+		if err != nil {
+			return err
+		}
+		if rel == ZarfYAML || rel == Checksums {
+			return nil
+		}
+		sum, err := helpers.GetSHA256OfFile(path)
+		if err != nil {
+			return err
+		}
+		checksumData = append(checksumData, fmt.Sprintf("%s %s", sum, filepath.ToSlash(rel)))
+		return nil
+	})
+	if err != nil {
+		return "", "", err
+	}
+	slices.Sort(checksumData)
+
+	checksumContent := strings.Join(checksumData, "\n") + "\n"
+	sha := sha256.Sum256([]byte(checksumContent))
+	return checksumContent, hex.EncodeToString(sha[:]), nil
+}
+
+func signPackage(dirPath, signingKeyPath, signingKeyPassword string) error {
+	if signingKeyPath == "" {
+		return nil
+	}
+	passFunc := func(_ bool) ([]byte, error) {
+		return []byte(signingKeyPassword), nil
+	}
+	keyOpts := options.KeyOpts{
+		KeyRef:   signingKeyPath,
+		PassFunc: passFunc,
+	}
+	rootOpts := &options.RootOptions{
+		Verbose: false,
+		Timeout: options.DefaultTimeout,
+	}
+	_, err := sign.SignBlobCmd(
+		rootOpts,
+		keyOpts,
+		filepath.Join(dirPath, ZarfYAML),
+		true,
+		filepath.Join(dirPath, Signature),
+		"",
+		false)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createReproducibleTarballFromDir(dirPath, dirPrefix, tarballPath string) error {
+	tb, err := os.Create(tarballPath)
+	if err != nil {
+		return fmt.Errorf("error creating tarball: %w", err)
+	}
+	defer tb.Close()
+
+	tw := tar.NewWriter(tb)
+	defer tw.Close()
+
+	// Walk through the directory and process each file
+	return filepath.Walk(dirPath, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		link := ""
```

Comment:
```suggestion
		var link string 
```

Reply:
Interesting matter of taste, I dont mind changing if this is what you prefer.